### PR TITLE
Remove adding unnecessary any cast when creating tuple values

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32892

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -5833,10 +5833,22 @@ public void visit(BLangTupleLiteral tupleLiteral) {
             result = rewriteExpr(typedescExpr);
             return;
         }
-        tupleLiteral.exprs.forEach(expr -> {
+        List<BLangExpression> exprs = tupleLiteral.exprs;
+        BTupleType tupleType = (BTupleType) tupleLiteral.getBType();
+        List<BType> tupleMemberTypes = tupleType.tupleTypes;
+        int tupleMemberExprSize = tupleMemberTypes.size();
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:5839]: Shall we say `tupleMemberTypeSize` instead?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -5833,10 +5833,22 @@ public void visit(BLangTupleLiteral tupleLiteral) {
...
+            BLangExpression expr = exprs.get(i);
             BType expType = expr.impConversionExpr == null ? expr.getBType() : expr.impConversionExpr.getBType();
-            types.setImplicitCastExpr(expr, expType, symTable.anyType);
-        });
+            BType targetType;
+            if (i < tupleMemberExprSize) {
+                targetType = tupleMemberTypes.get(i);
+            } else {
+                targetType = tupleType.restType;
+            }
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:5849]: Can use a ternary operator here also?

**Review by MaryamZi — COMMENTED** [review]
Even with this fix, if every (or most) members need to be boxed, it will still result in the error, right?

Shall/can we also add tests?

**Comment by warunalakshitha** [comment]
> Even with this fix, if every (or most) members need to be boxed, it will still result in the error, right?
> 
> Shall/can we also add tests?

I can add tests but wondering do we need? . Existing tests should cover the code. If members need to box it will box since types.setImplicitCastExpr(expr, expType, targetType) will handle it.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**